packages/flutter_tools/lib/src/commands/doctor.dart

```diff
@@ -26,6 +32,14 @@ class DoctorCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    final String engineRevision = argResults['check-for-remote-artifacts'];
```

Comment:
You might check the engine revision to make sure it's a hash, and that it's not empty (i.e. compare against hexadecimal numbers regex), something like this (and only check for lowercase hashes):

```dart
    if (argResults.wasParsed('check-for-remote-artifacts')) {
      final String engineRevision = argResults['check-for-remote-artifacts'].toLowerCase();
      if (engineRevision.startsWith(RegExp(r'[a-f0-9]{1,40}'))) {
        final bool success = await doctor.checkRemoteArtifacts(engineRevision);
        if (!success) {
          throwToolExit(
              'Artifacts for engine $engineRevision are missing or are not yet available.',
              exitCode: 1);
        }
      } else {
        throwToolExit('Remote artifact revision "$engineRevision" is not a valid git hash.');
      }
    }
```

Reply:
Done.